design/object-metadata-s3.md

```diff
@@ -0,0 +1,48 @@
+# Object Metadata and S3
+
+### Overview
+
+lakeFS keeps the following metadata per object:
+
+1. Physical address
+1. Last modified time
+1. Size
+1. ETag
+1. Metadata
+1. Address type (relative / full)
+
+### What is missing?
+
+Mapped two issues that we currently do not address when we work with our S3 interface:
+
+1. Content-Type - Not keeping the posted object content-type and do not set it back when we get the object.
+    [Support Content-Type](https://github.com/treeverse/lakeFS/issues/2296)
+    We saw this issue when one of the clients used the value of the content type to identify a directory marker. But this is one of the basic properties that evey object store manages.
+1. Additional metadata - when posting an object aws enables additional metadata to set. Currently we do not process these attributes. The object level metadata we currenly have on an object can be get/set only by our open api.
+    [Store some per-file metadata (rclone multipart upload fails on unexpected ETag) #2486](https://github.com/treeverse/lakeFS/issues/2486)
+
+
+### Solution
+
+AWS metadata posted with an object will be added to the object's metadata. We will map "x-amz-meta-<name>" to name/value in our metadata. The equivalent to create entry metadata using our open api.
+The s3 gateway get object will map the metadata key/value back to "x-amz-meta-<name>" headers.
+
+The content-type and the new metadata posted in multipart upload will be stored in our tracker. This information should be kept until the multipart complete is called and then the metadata is stored.
+
+In order to support content-type for all old entries, the loaded entries will be set with default content-type if found missing. This will not be reflected on entry until an update to entry will be made, as the data is alreay committed, but any future commit that include the object will also update the content-type.
+
+Content-type will be managed as part of the entry's Metadata field. The catalog will make sure the default is set in case no value or empty value is set.
+
+Example of how head request on object with content-type will look like:
+
+```json
+{
+   "AcceptRanges": "bytes",
+   "LastModified": "Tue, 28 Sep 2021 22:34:44 GMT",
+   "ContentLength": 10485760,
+   "ETag": "\"f962bf40d19fed2e80bcbaa33bd1dfe7\"",
+   "ContentType": "example/data",
+   "Metadata": {}
```

Comment:
Do I (additionally) get the metadata here, or only in the `x-amz-meta-*` headers?  If not here, why do we have this field?